Add profile article for shell experience (#9069)
MicrosoftDocs/PowerShell-Docs: reference/docs-conceptual/learn/shell/creating-profiles.md

---
description: >
  This article explains how to use your profile to save preferred PowerShell settings and optimize
  your shell experience.
ms.date: 07/28/2022
title: Customizing your shell environment
---
# Customizing your shell environment

A PowerShell profile is a script that runs when PowerShell starts. You can use the profile to

> customize the environment. You can add aliases, functions, variables, load modules, create
> PowerShell drives, and change preference settings. Putting these settings in your profile ensures
> they are available whenever you start PowerShell on your system.

Reviewer: I don't love this, but I wanted to add `run commands` explicitly and the construction of `add aliases, functions, variables, load modules...` didn't sit quite right.

```suggestion
customize the environment. You can:

- add aliases, functions, and variables
- load modules
- create PowerShell drives
- run arbitrary commands
- and change preference settings.

Putting these settings in your profile ensures they are available whenever you start PowerShell on
your system.
```

## The $PROFILE variable

The `$PROFILE` automatic variable stores the paths to the PowerShell profiles that are available in
the current session.

There are four possible profiles available to support different user scopes and different PowerShell
hosts. The fully qualified paths for each profile script are stored in the following member
properties of `$PROFILE`.

- **AllUsersAllHosts**
- **AllUsersCurrentHost**
- **CurrentUserAllHosts**
- **CurrentUserCurrentHost**

> You can see, you can create profile scripts that run for all users or just one user, the
> **CurrentUser**. **CurrentUser** profiles are stored in the user's home directory.

Reviewer: ```suggestion
You can create profile scripts that run for all users or just one user, the **CurrentUser**.
**CurrentUser** profiles are stored in the user's home directory.
```

There are also profiles that run for all PowerShell hosts or specific hosts. The profile script
for each PowerShell host has a name unique for that host. For example, the filename for the standard
Console Host on Windows or the default terminal application on other platforms is
`Microsoft.PowerShell_profile.ps1`. For Visual Studio Code (VS Code), the filename is
`Microsoft.VSCode_profile.ps1`.

For more information, see [about_Profiles][1].

By default, referencing the `$PROFILE` variable returns the path to the "Current User, Current Host"
profile. The other profiles path can be accessed through the properties of the `$PROFILE` variable.
For example:

```powershell
PS> $PROFILE
C:\Users\user1\Documents\PowerShell\Microsoft.PowerShell_profile.ps1
PS> $PROFILE.AllUsersAllHosts
C:\Program Files\PowerShell\7\profile.ps1
```

## How to create your personal profile

> When you first install PowerShell on a system, profile script and the directory path it belong is
> don't exist. The following command creates the profile script file if the file doesn't exist.

Reviewer: ```suggestion
When you first install PowerShell on a system, the profile script files and the directories they
belong to don't exist. The following command creates the "Current User, Current Current Host"
profile script file if it doesn't exist.
```
